_inc/client/components/mobile-magic-link/index.jsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import { translate as __ } from 'i18n-calypso';
+import analytics from 'lib/analytics';
+import Card from 'components/card';
+import Button from 'components/button';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getSiteConnectionStatus as _getSiteConnectionStatus,
+	isCurrentUserLinked as _isCurrentUserLinked,
+} from 'state/connection';
+import Modal from 'components/modal';
+import onKeyDownCallback from 'utils/onkeydown-callback';
+import { sendMobileLoginEmail } from 'state/mobile/actions';
+
+export class MobileMagicLink extends React.Component {
+	static displayName = 'MobileMagicLink';
+
+	state = {
+		showModal: false,
+	};
+
+	handleOpenModal = e => {
+		analytics.tracks.recordJetpackClick( 'connect_mobile_app' );
+		e.preventDefault();
+		this.toggleModalVisibility();
+	};
+
+	toggleModalVisibility = () => {
+		this.setState( {
+			showModal: ! this.state.showModal,
+		} );
+	};
+
+	clickSendLoginEmail = e => {
+		e.preventDefault();
+		this.toggleModalVisibility();
+		this.props.sendMobileLoginEmail();
+	};
+
+	render() {
+		const { showModal } = this.state;
+		return (
+			<div className="mobile-magic-link">
+				<a
+					className="mobile-magic-link__button"
+					onClick={ this.handleOpenModal }
+					onKeyDown={ onKeyDownCallback( this.handleOpenModal ) }
+					role="button"
+					tabIndex="0"
+				>
+					{ __( 'Connect to mobile WordPress app' ) }
+				</a>
+				{ showModal && (
+					<Modal className="mobile-magic-link__modal" onRequestClose={ this.toggleModalVisibility }>
+						<Card className="mobile-magic-link__modal__body">
+							<h2>{ __( 'Email me a link to log in the app' ) }</h2>
+							<h4>
+								{ __(
+									"Easily log in to the WordPress.com app by clicking the link we'll send to the email address on your account."
```

Comment:
`Easily log in to the WordPress.com app by clicking [...]`
Probably we don't want to say "WordPress.com".  How about just "WordPress".

Reply:
Maybe:
"Easily log in to the WordPress mobile app with a special link sent to your email address.  Just click the link from your mobile device to log in!"  

Dunno. Words are hard.

Reply:
> Dunno. Words are hard.

Heard that. 😂

Reply:
> Words are hard.

We (non native speakers) agree!

![_97587522_9ea23dbd-7ff4-4228-9f5c-94824ed857fc](https://user-images.githubusercontent.com/348248/59115322-a9090980-890e-11e9-8818-1ca75769ee55.jpg)